File: 003/prime-factor.js
let n = 600851475143
let limit = Math.ceil(Math.sqrt(n))
for (var i = 3; i <= limit; i += 2) {
  while (n % i === 0) {
    n = Math.floor(n / i)
    limit = Math.ceil(Math.sqrt(n))
  }
}
console.log(n)

File: 003/prime-factor.rs
fn main() {
    let mut n: u64 = 600851475143;
    let mut limit = n / 2;
    let mut i = 3;
    while i < limit {
        while n % i == 0 {
            n = n / i;
            limit = n / 2;
        }
        i += 2;
    }
    println!("{}", n);
}
